### ISSUE TITLE
Compilation issue on Fedora

### ISSUE THREAD

**Issue by eeickmeyer** [monocasual/giada#321]
Hi there. I'm the new maintainer of giada in Fedora. Looks like it went several versions without an update, so I'm here to fix that.

I was alerted to the 0.16.2.2 tag before I went to bed, and decided to work on it when I woke up this morning. However, the issue I'm running into is this:

`src/core/midiMapConf.cpp:33:10: fatal error: deps/json/single_include/nlohmann/json.hpp: No such file or directory`

I've got all the dependencies, but I cannot resolve this one as I have no idea what, if anything, can be patched here as I cannot find what it's looking for in your tree.

If you could help me resolve this, it'd be much appreciated. :)

**Comment by napcode** [comment]
Hi,

did you follow [this](https://www.giadamusic.com/documentation-compiling-from-source)? Looks like it is missing one of the submodules. Does the folder src/deps/json exist?

**Comment by dvzrv** [comment]
@eeickmeyer this is more or less a duplicate of #315.

TL;DR: This repository has submodules and they are not included in the auotmatically generated source tarball. Amongst a modified version of rtaudio, there's also nlohmann-json.

**Comment by monocasual** [comment]
HI @eeickmeyer , as @dvzrv notes this is the same problem of #315. I'd really love to fix this in a nice and clean way as stated [here](https://github.com/monocasual/giada/issues/315#issuecomment-592933248). In the meantime, @dvzrv will kindly provide a script to create a source tarball with all the dependencies included.

**Comment by eeickmeyer** [comment]
That would be absolutely excellent. One of the hard things about packaging is having to repack for submodules, which is a gray area as it creates a tarball that isn't directly from upstream. :)
